### PR TITLE
Add OpenSky demo

### PR DESCRIPTION
**Pull request by hpritchett-dstl**
Add demonstration of a Stone Soup multi-target tracker applied to ADS-B data taken from The OpenSky Network.

### CODE REVIEW

File: docs/demos/OpenSky_Demo.py [hunk 1]
```diff
@@ -0,0 +1,637 @@
...
+
+Fullscreen().add_to(m)
+
+(TimestampedGeoJson(
+    data=geo_json,
+    transition_time=200,
+    auto_play=True,
+    add_last_point=False,
+    period='PT10S',
+    duration='PT0S')).add_to(m)
```
Comment (sdhiscocks) [inline docs/demos/OpenSky_Demo.py:631]:
```suggestion
TimestampedGeoJson(
    data=geo_json,
    transition_time=200,
    auto_play=True,
    add_last_point=False,
    period='PT10S',
    duration='PT0S').add_to(m)
```

File: docs/demos/OpenSky_Demo.py [hunk 2]
```diff
@@ -0,0 +1,570 @@
...
+# The update step calculates the posterior state estimate by using both our prediction and
+# sensor measurement. We don't need to define the measurement model here since the
+# model :class:`~.CartesianToBearingRange` is already provided in the measurements.
+#
+# The :class:`~.DistanceHypothesiser` generates track predictions at detection times, and
+# scores each hypothesised prediction-detection pair using our set measure of :class:`~.Mahalanobis` distance.
+# We allocate the detections to our predicted states by using the Global Nearest Neighbour method.
+# The :class:`~.UpdateTimeDeleter` will identify the tracks for deletion and delete them once
+# the time since last update has exceeded our specified time.
+# By having delete_last_pred = True, the state that caused a track to be deleted will be deleted
```
Comment (nperree-dstl) [inline docs/demos/OpenSky_Demo.py:177]:
```suggestion
# By having `delete_last_pred = True`, the state that caused a track to be deleted will be deleted
```

File: docs/demos/OpenSky_Demo.py [hunk 3]
```diff
@@ -0,0 +1,570 @@
...
+        all_truths[id_]['times'].append(time)
+        all_truths[id_]['lonlats'].append((lon, lat))
+        all_truths[id_]['velocity'].append(truth.metadata.get('velocity'))
+        all_truths[id_]['heading'].append(truth.metadata.get('heading'))
+
+# %%
+# A list of all icao24 addresses in our data will also be created. These addresses are used to
+# give an aircraft a unique identity. We will run through this list and use our groundtruth
+# dictionary to get relevant data needed for plotting.
+# The variable trail_Size will determine how much history of the track we want to be seen.
```
Comment (nperree-dstl) [inline docs/demos/OpenSky_Demo.py:463]:
```suggestion
# The variable `trail_Size` will determine how much history of the track we want to be seen.
```

File: docs/demos/OpenSky_Demo.py [hunk 4]
```diff
@@ -0,0 +1,570 @@
...
+# We will plot a LineString to display our track, taking into account our specified trail_Size
```
Comment (nperree-dstl) [inline docs/demos/OpenSky_Demo.py:464]:
```suggestion
# We will plot a LineString to display our track, taking into account our specified `trail_Size`
```

**Review by nperree-dstl — APPROVED** [review]
Just a couple of formatting points to make it read nicer and a fix for a broken link. Otherwise looks really good!

**Review by sdhiscocks — CHANGES_REQUESTED** [review]
Looks good. The only thing is the transition model noise parameters look very large. I think this is an issue with the initial prior for velocity being too small. Suggested patch below, that increases the prior uncertainty, which then means model uncertainty can be reduced to much lower values.

```patch
diff --git a/docs/demos/OpenSky_Demo.py b/docs/demos/OpenSky_Demo.py
index d1aee11d..e23558aa 100644
--- a/docs/demos/OpenSky_Demo.py
+++ b/docs/demos/OpenSky_Demo.py
@@ -182,9 +182,9 @@ detection_sim = PlatformDetectionSimulator(groundtruth, platforms)
 # (if it is a prediction).
 
 transition_model = CombinedLinearGaussianTransitionModel(
-    (ConstantVelocity(1500), ConstantVelocity(1500), ConstantVelocity(1000)))
+    (ConstantVelocity(20), ConstantVelocity(20), ConstantVelocity(2)))
 init_transition_model = CombinedLinearGaussianTransitionModel(
-    (ConstantVelocity(8500), ConstantVelocity(8500), ConstantVelocity(1000)))
+    (ConstantVelocity(5), ConstantVelocity(5), ConstantVelocity(2)))
 
 from stonesoup.predictor.kalman import ExtendedKalmanPredictor
 predictor = ExtendedKalmanPredictor(transition_model)
@@ -195,7 +195,7 @@ updater = ExtendedKalmanUpdater(measurement_model=None)
 
 from stonesoup.hypothesiser.distance import DistanceHypothesiser
 from stonesoup.measures import Mahalanobis
-hypothesiser = DistanceHypothesiser(predictor, updater, Mahalanobis(), missed_distance=3)
+hypothesiser = DistanceHypothesiser(predictor, updater, Mahalanobis(), missed_distance=5)
 
 from stonesoup.dataassociator.neighbour import GNNWith2DAssignment
 data_associator = GNNWith2DAssignment(hypothesiser)
@@ -216,11 +216,11 @@ from stonesoup.types.state import GaussianState
 initiator = MultiMeasurementInitiator(
     GaussianState(
         np.array([[0], [0], [0], [0], [0], [0]]),   # Prior State
-        np.diag([8000, 90, 8000, 90, 0, 20])),
+        np.diag([15**2, 100**2, 15**2, 100**2, 15**2, 20**2])),
     measurement_model=None,
     deleter=deleter,
     data_associator=GNNWith2DAssignment(
-        DistanceHypothesiser(init_predictor, updater, Mahalanobis(), missed_distance=1.8)),
+        DistanceHypothesiser(init_predictor, updater, Mahalanobis(), missed_distance=3)),
     updater=updater,
     min_points=2
     )

```

**Review by sdhiscocks — APPROVED** [review]

**Pull request merged**